distributed/node.py

```diff
@@ -131,12 +131,9 @@ def start_http_server(
             import ssl
 
             ssl_options = ssl.create_default_context(
-                cafile=tls_ca_file, purpose=ssl.Purpose.SERVER_AUTH
+                cafile=tls_ca_file, purpose=ssl.Purpose.CLIENT_AUTH
```

Comment:
From https://docs.python.org/3/library/ssl.html#ssl.Purpose.SERVER_AUTH

> Purpose.SERVER_AUTH
>
> Option for [create_default_context()](https://docs.python.org/3/library/ssl.html#ssl.create_default_context) and [SSLContext.load_default_certs()](https://docs.python.org/3/library/ssl.html#ssl.SSLContext.load_default_certs)[](https://docs.python.org/3/library/ssl.html#ssl.Purpose.CLIENT_AUTH). This value indicates that the context may be used to authenticate web servers (therefore, it will be used to create client-side sockets).
>
> Purpose.CLIENT_AUTH
>
> Option for [create_default_context()](https://docs.python.org/3/library/ssl.html#ssl.create_default_context) and [SSLContext.load_default_certs()](https://docs.python.org/3/library/ssl.html#ssl.SSLContext.load_default_certs). This value indicates that the context may be used to authenticate web clients (therefore, it will be used to create server-side sockets).

I understand this change removes the `ssl.SSLError` but as I don't have a good grasp of SSL and security, I think I should nit here and ask what this change does in greater detail.

Reply:
`HTTPServer` creates server-side sockets and so needs an SSLContext of the `ssl.Purpose.CLIENT_AUTH` kind. so this should always have been `purpose=ssl.Purpose.CLIENT_AUTH`

also CLIENT_AUTH sets
```python
            ssl_options.check_hostname = False
            ssl_options.verify_mode = ssl.CERT_NONE
```
already so we don't need those overrides anymore